DatadogCore/Sources/Datadog.swift

```diff
@@ -346,14 +361,14 @@ public struct Datadog {
         // Set default `DatadogCore`:
         let core = DatadogCore(
             directory: try CoreDirectory(
-                in: Directory.cache(),
+                in: configuration.systemDirectory(),
                 instancenName: instanceName,
                 site: configuration.site
             ),
             dateProvider: configuration.dateProvider,
             initialConsent: trackingConsent,
             performance: performance,
-            httpClient: HTTPClient(proxyConfiguration: configuration.proxyConfiguration),
+            httpClient: configuration.httpClientFactory(configuration.proxyConfiguration),
```

Comment:
Following my comment above:
```suggestion
            httpClient: URLSessionClient(proxyConfiguration: configuration.proxyConfiguration),
```